daemon/create.go

```diff
@@ -144,5 +144,9 @@ func (daemon *Daemon) VolumeCreate(name, driverName string, opts map[string]stri
 	if err != nil {
 		return nil, err
 	}
+
+	if driverName != "" && v.DriverName() != driverName {
+		return nil, fmt.Errorf("cannot create volume '%s' using the '%s' driver because the volume '%s' already exists and the driver name is '%s'", name, driverName, name, v.DriverName())
```

Comment:
error message should start with `Conflict`, and we should add the error response to the API docs.